searchlib/src/vespa/searchlib/features/attributefeature.cpp

```diff
@@ -277,18 +278,19 @@ AttributeBlueprint::AttributeBlueprint() :
 AttributeBlueprint::~AttributeBlueprint() = default;
 
 void
-AttributeBlueprint::visitDumpFeatures(const search::fef::IIndexEnvironment &,
-                                      search::fef::IDumpFeatureVisitor &) const
+AttributeBlueprint::visitDumpFeatures(const fef::IIndexEnvironment &,
+                                      fef::IDumpFeatureVisitor &) const
 {
 }
 
 bool
-AttributeBlueprint::setup(const search::fef::IIndexEnvironment & env,
-                          const search::fef::ParameterList & params)
+AttributeBlueprint::setup(const fef::IIndexEnvironment & env,
+                          const fef::ParameterList & params)
 {
     // params[0] = attribute name
     // params[1] = index (array attribute) or key (weighted set attribute)
     _attrName = params[0].getValue();
+    _attrKey = getBaseName() + "." + _attrName;
```

Comment:
if the intent is to share attribute lookups between features, consider using a name explicitly specific for the attribute itself and not for this feature/attribute combination. (NB: beware of name collisions)

Reply:
Intent is to across features. Moved to base class.